WordPress/Classes/Utility/WPURLRequest.m

```diff
@@ -21,14 +21,19 @@ + (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
     NSParameterAssert(username);
     NSParameterAssert(password != nil || bearerToken != nil);
     
-    NSMutableURLRequest *request = [self mutableRequestWithURL:loginUrl userAgent:userAgent];
-
-    NSString *hostname = [loginUrl host];
+    NSString *hostname = loginUrl.host;
+    
+    // Let's make sure we don't send OAuth2 tokens outside of wordpress.com
     if (![hostname isEqualToString:@"wordpress.com"] && ![hostname hasSuffix:@".wordpress.com"]) {
-        // Let's make sure we don't send OAuth2 tokens outside of wordpress.com
         bearerToken = nil;
+        
+    // Enforce HTTPS for WordPress.com Sites
+    } else if ([loginUrl.scheme isEqual:@"http"]) {
+        loginUrl = [[NSURL alloc] initWithScheme:@"https" host:loginUrl.host path:loginUrl.path];
```

Comment:
Careful with this... this does not maintain the query field of a URL.  I've tested with the following code and the resulting URL is not the same:

```objective-c
    NSURL *url = [NSURL URLWithString:@"http://diego.com/testing/path?no=joking&is=true"];
    NSURL *url2 = [[NSURL alloc] initWithScheme:url.scheme host:url.host path:url.path];
```